set is a default gem in ruby 3.x still, so don't depend on the set gem

Index: .metadata
--- .metadata.orig
+++ .metadata
@@ -11,20 +11,6 @@ cert_chain: []
 date: 2021-02-13 00:00:00.000000000 Z
 dependencies:
 - !ruby/object:Gem::Dependency
-  name: set
-  requirement: !ruby/object:Gem::Requirement
-    requirements:
-    - - "~>"
-      - !ruby/object:Gem::Version
-        version: '1.0'
-  type: :runtime
-  prerelease: false
-  version_requirements: !ruby/object:Gem::Requirement
-    requirements:
-    - - "~>"
-      - !ruby/object:Gem::Version
-        version: '1.0'
-- !ruby/object:Gem::Dependency
   name: rbtree
   requirement: !ruby/object:Gem::Requirement
     requirements:
